spec/fixtures/apply/basic/plans/init.pp

```diff
@@ -7,9 +7,11 @@
   return apply($nodes) {
     file { '/root/test/':
       ensure => directory,
-    } -> file { "/root/test/hello.txt":
-      ensure => file,
+    } -> file { '/root/test/hello.txt':
+      ensure  => file,
       content => "hi there I'm ${$facts['os']['family']}\n",
     }
+  }.map |$r| {
```

Comment:
What is the purpose of this? It seems like there is no key `$r['resources']` so `null` is returned. A key `resources` is nested under `metrics`. `$r['metrics']['resources']`

Reply:
That's surprising. What we should get back from `apply` (in these tests) is the catalog produced from compiling the manifest block. It iterates over the result of each node, and expects to access the `resources` property of each catalog.

Running the plan directly won't work, it has to be run with apply_catalog.rb overridden by `spec/fixtures/apply/apply_catalog.sh`.

Reply:
Ahhh, I see. I was running as a plan outside of the test.